snapshots/overlay/plugin/plugin.go

```diff
@@ -57,6 +57,13 @@ func init() {
 				oOpts = append(oOpts, overlay.WithUpperdirLabel)
 			}
 
+			if config.Metacopy != nil {
+				oOpts = append(oOpts, overlay.WithMetacopyCtrl)
+				if *config.Metacopy {
+					oOpts = append(oOpts, overlay.WithMetacopy)
+				}
+			}
```

Comment:
I'm not convinced still that making the field in `SnapshotterConfig` a `*bool` isn't simpler. We just assign it here (we won't need two "overlay.WithMetacopy..." functions), and then we test for not nil in other parts of the code (instead of metacopyCtr) and that is it.

Just my personal taste (and I'm no maintainer here), definitely not a must.

Reply:
Here I agree, but I just don't want to change the original behavior. Currently the parameters for `func NewSnapshotter(root string, opts ...Opt) (snapshots.Snapshotter, error)` function are passed using "ops", like
```
// WithUpperdirLabel adds as an optional label
// "containerd.io/snapshot/overlay.upperdir". This stores the location
// of the upperdir that contains the changeset between the labelled
// snapshot and its parent.
func WithUpperdirLabel(config *SnapshotterConfig) error {
    config.upperdirLabel = true
    return nil
}
```
We can pass SnapshotterConfig structure directly to NewSnapshotter, but in this case it should be prepared in plugin.go.
I can do this, not a problem, if there no objections from other reviewers, due to I don't know the story, why current approach was used.